azure-quantum/azure/quantum/providers/ionq.py

```diff
@@ -0,0 +1,75 @@
+import io
+import json
+from typing import Any, Dict, Tuple
+
+from azure.quantum import Job
+from azure.quantum._client.models import JobDetails
+from azure.quantum.storage import upload_blob, ContainerClient
+from azure.quantum.workspace import Workspace
+
+IonQJson = Dict[str, Any]
+IONQ_INPUT_DATA_FORMAT = "ionq.circuit.v1"
+IONQ_OUTPUT_DATA_FORMAT = "ionq.quantum-results.v1"
+IONQ_PROVIDER_ID = "IonQ"
+DEFAULT_TIMEOUT = 100
+
+
+class IonQ:
+    def __init__(self, workspace: Workspace, target: str = "ionq.simulator"):
+        self.workspace = workspace
+        self.target = target
+    
+    def submit(
```

Comment:
I was hoping to push most of this functionality to the Job class itself. For example uploading the Blob and populating the JobDetails.
Then we can clean the ../optimization/Solver code to use the same logic, so we don't have that logic duplicated.